lib/storage/metadata/mongoclient/LogConsumer.js

```diff
@@ -89,18 +91,17 @@ class LogConsumer {
             if (err) {
                 return cb(err);
             }
-            return this._coll.find({
+            const cursor = this._coll.find({
                 ns: this._oplogNsRegExp,
             }, {
                 tailable: true,
                 awaitData: true,
                 noCursorTimeout: true,
                 numberOfRetries: Number.MAX_VALUE,
-            }, (err, cursor) => {
-                const recordStream = new ListRecordStream(
-                    cursor, this._logger, startSeq.uniqID, latestOplogID);
-                return cb(null, { log: recordStream, tailable: true });
             });
+            const recordStream = new ListRecordStream(
```

Comment:
should `ListRecordStream` not be updated as well, to use the new custer?
looking at https://github.dev/scality/Arsenal/blob/dd9283d156a5d7fa57987e48314aa3f904f4508a/lib/storage/metadata/mongoclient/ListRecordStream.js#L47, we see it is still using the callback-based API...

Reply:
Yes, I can include it in this PR as well...  I add some commits.

Reply:
Done in https://github.com/scality/Arsenal/pull/2351/commits/44f4beaa759ef9d62968c9e137a23e74927535e3

Reply:
This whole logic will be removed in ARSN-486.